lib/system/alloc.nim

```diff
@@ -943,12 +943,13 @@ proc rawDealloc(a: var MemRegion, p: pointer) =
                 s -% sizeof(FreeCell))
       # check if it is not in the freeSmallChunks[s] list:
       if c.free < s:
+        # TODO: Steal freeList from the other chunk? Leaving it as is will waste a lot of cells until we return to the other chunk
```

Comment:
What does that mean? Do we leak here or is the memory reused later?

Reply:
The memory should be reused _eventually_, but through bad ordering of dealloc a chunk may hold onto both owned and foreign cells until it becomes the head of the list again.

Reply:
It is also difficult to free a suspended chunk because of the potential for shared cells, so it must stay in memory.

Reply:
Because of similar reasons I was thinking that a chunk doesn’t actually need the `free` field, but no concrete ideas to improve it.

Reply:
Looking at it again the `TODO` probably isn't a concern because it should cycle the chunks.